test/mocks.js

```diff
@@ -64,5 +135,8 @@ function mockObject (methodNames) {
 module.exports = {
   mockLog: mockLog,
   spyLog: spyLog,
+  createDB: createDB,
+  createMailer: createMailer,
+  createRequest: createRequest,
```

Comment:
Nice refactor and cleanup on this mocking code!  Is it worth using the new `mockObject` helper here to make them even more flexible?

Naming nit: for consistency with the existing exports, should these be something `mockDB`, `mockMailer`, etc?